docs/reference/pages/model_reference.md

```diff
@@ -738,6 +738,31 @@ Every time a page is edited, a new `Revision` is created and saved to the databa
+
+    Note: Purging Stale Revisions
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Stale revisions are revisions that are no longer associated with any live objects. These revisions can accumulate over time as content is updated and new revisions are created. To manage database size and improve performance, administrators can use the `purge_revisions` management command to remove stale revisions from the database.
+    Please exercise caution when running the `purge_revisions` command, as it will permanently delete stale revisions that are no longer referenced by any live content. Make sure to back up important revisions or consider using a ForeignKey with `on_delete=models.PROTECT` if there are specific revisions that you want to preserve and prevent from being deleted.
+    Using ForeignKey with `on_delete=models.PROTECT`
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Wagtail implementors may want to use a `ForeignKey` field with `on_delete=models.PROTECT` when referring to a specific revision they don't want to be deleted. By doing so, the revision will be protected from deletion as long as it is referenced by any other live objects in the system.
+    Stale revisions are revisions that are no longer associated with any live objects. These revisions can accumulate over time as content is updated and new revisions are created. To manage database size and improve performance, administrators can use the `purge_revisions` management command to remove stale revisions from the database.
+    Please exercise caution when running the `purge_revisions` command, as it will permanently delete stale revisions that are no longer referenced by any live content. Make sure to back up important revisions or consider using a ForeignKey with `on_delete=models.PROTECT` if there are specific revisions that you want to preserve and prevent from being deleted.
+    Using ForeignKey with `on_delete=models.PROTECT`
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Wagtail implementors may want to use a `ForeignKey` field with `on_delete=models.PROTECT` when referring to a specific revision they don't want to be deleted. By doing so, the revision will be protected from deletion as long as it is referenced by any other live objects in the system.
+
+    Here's an example of using `on_delete=models.PROTECT` with a `ForeignKey` field in your model definition:
+    .. code-block:: python
+        from django.db import models
+        from wagtail.core.models import Revision
+        class MyModel(models.Model):
+            revision = models.ForeignKey(
+                Revision,
+                on_delete=models.PROTECT,
+                )
+            # Add other fields for your model
+    With this setup, when a `Revision` instance is referenced by the `revision` field of `MyModel`, the `purge_revisions` command will not delete that particular revision, ensuring its preservation even if it becomes stale.
```

Comment:
There's a lot of repetitions in these paragraphs and they read a bit AI-generated to me. Also, instead of referring to "Wagtail implementors" or "administrators", just use "you" because this feature is intended for developers and we're writing developer docs.

That said, I don't think this explanation should be in the `Revision` model reference part of the documentation. I'd suggest moving this to the documentation for the `purge_revisions` command: https://github.com/wagtail/wagtail/blob/a2c9e9ab2cf3e4f0808c340f8df94882de82b5f3/docs/reference/management_commands.md?plain=1#L45

And add a new paragraph/sentence after the bullet points in the `Revision` brief to link to the `purge_revisions` docs:

https://github.com/wagtail/wagtail/blob/a2c9e9ab2cf3e4f0808c340f8df94882de82b5f3/docs/reference/pages/model_reference.md?plain=1#L676-L682

Reply:
Should I remove the entire thing from the pages reference or only parts of it? It did seem kinda out of place there.

Reply:
I'd suggest removing it entirely and write a sentence e.g.

```markdown
You can use the [`purge_revisions`](purge_revisions) command to delete old revisions that are no longer in use.
```

to the `Revision` brief I mentioned above.

Reply:
I've made changes accordingly in [f237d2e](https://github.com/wagtail/wagtail/pull/10961/commits/f237d2e2944784feea8d6a8576e8ad4c17c38e16). Can you please verify them?

Reply:
I'm not sure if this:
```
[`purge_revisions`](https://github.com/wagtail/wagtail/blob/a2c9e9ab2cf3e4f0808c340f8df94882de82b5f3/docs/reference/management_commands.md?plain=1#L45)
```
is the correct way to link to the purge_revisions subheading in the management_commands.md

Reply:
You can copy-paste the block above as-is and the link should work because we've added an explicit target for it:
https://github.com/wagtail/wagtail/blob/953c980976231d6d2e1d26a58595710dcf10cc71/docs/reference/management_commands.md?plain=1#L43-L45

https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html

Run `make livehtml` inside the `docs` directory to build the docs locally.

Reply:
I was wondering what that was for. That's a pretty cool functionality!

Reply:
Can you please verify the changes in [1a173af](https://github.com/wagtail/wagtail/pull/10961/commits/1a173af7277676f8bed4548f9515c99b0cfbe808)?